pom.xml

```diff
@@ -19,100 +19,93 @@
   <parent>
     <groupId>org.sonatype.nexus.plugins</groupId>
     <artifactId>nexus-plugins</artifactId>
-    <version>3.18.0-01</version>
+    <version>3.19.0-SNAPSHOT</version>
   </parent>
 
-  <artifactId>nexus-repository-helm</artifactId>
+  <artifactId>nexus-repository-base</artifactId>
+  <groupId>org.sonatype.nexus.plugins</groupId>
   <name>${project.groupId}:${project.artifactId}</name>
-  <version>0.0.12</version>
+  <version>1.0.0</version>
```

Comment:
Since I don't think we have a reason for different versions, I would keep the version defined in the parent only (and do not define a <version> tag in sub-module poms - thereby allowing the sub-modules to inherit the version from the parent/base).

Oh, wait. Now I see how you're attempting to define the version in the sub-module pom. Hm...Not sure the will work: When we do a release of a multi-module project, we will also be releasing the "parent"/"base" pom.xml. I think the parent module with then require a unique version too, and if so, we're back to having to update the parent/base version (and sub-module parent tags) anyway...

I'm not certain of the above, and I'm willing to try your approach if you think my concern can be addressed.

Reply:
Not sure that I understand your concerns. 
`nexus-repository-base` parent module has version defined `1.0.0`
Submodule `nexus-repository-helm` refers to the parent by its version `1.0.0` and has it's own version defined in the parent pom `<helm-plugin-version>0.0.13</helm-plugin-version>`. This allows us to release submodule and use the new version in the scope of one parent module release.

Reply:
I'm fairly certain this versioning as you have it now will need to be changed.

* We do not have use cases where CI can deploy only a submodule, and we should avoid any such case if one arrises. The ITs and Format module should typically be released at the same time. Better to release a now version of all modules (parent, format, it).

* The property that defines the version of the sub-modules is defined in the parent, and therefore you would have to alter the parent module (edit that property) to update and release a sub-module. This edit requires the parent module get a new version too - maven does not allow redeploy of the same versions of an artifact.

* The CircleCI [ORB](https://circleci.com/orbs/registry/orb/github-maven-deploy/github-maven-deploy) we use should handle updating the parent and sub-module versions just fine (including the <parent> tag in sub-module) due to use of the version helper plugin->versions:set command:

```
          internal_versioncommand: mvn build-helper:parse-version versions:set -DnewVersion=\${parsedVersion.majorVersion}.\${parsedVersion.minorVersion}.\${parsedVersion.nextIncrementalVersion} versions:commit
```